lib/reek/smell_detectors/module_initialize.rb

```diff
@@ -21,14 +21,18 @@ def self.contexts # :nodoc:
       # @return [Array<SmellWarning>]
       #
       def sniff
-        context.defined_instance_methods.each do |node|
-          if node.name == :initialize
-            return smell_warning(
-              lines:   [source_line],
-              message: 'has initialize method')
-          end
-        end
-        []
+        return [] if does_not_have_initializer_method?
```

Comment:
@mvz @troessner I'd like to know what you think about the naming. thanks!

Reply:
I am fine with the naming :)